client/src/components/Tool/ToolForm.vue

```diff
@@ -335,17 +337,21 @@ export default {
                             toolName: this.toolName,
                         };
                         this.saveLatestResponse(response);
-                        this.$router.push(`/jobs/submission/success`);
+                        changeRoute = prevRoute === this.$route.fullPath;
                     } else {
                         this.showError = true;
                         this.showForm = true;
                         this.errorTitle = "Job submission rejected.";
                         this.errorContent = jobResponse;
                     }
-                    if ([true, "true"].includes(config.enable_tool_recommendations)) {
-                        this.showRecommendation = true;
+                    if (changeRoute) {
```

Comment:
Did you compare this approach to using route guards?

What you have here seems like it might actually be a better solution in that it keeps the logic specific to this promise local, just curious if you poked at the other option.

Reply:
Oh I am not aware of those, I can look into it and try it out tomorrow